<!--
 Copyright 2023 Shadow Robot Company Ltd.
 This program is free software: you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the Free
 Software Foundation version 2 of the License.
 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 more details.
 You should have received a copy of the GNU General Public License along
 with this program. If not, see <http://www.gnu.org/licenses/>.
-->
<launch>
    <!-- Boolean stating whether the object should be deleted -->
    <arg name="delete_object" default="false"/>
    <!-- Name you want to give to the object -->
    <arg name="object_name" default=""/>
    <!-- Type of the object to spawn -->
    <arg name="object_type" default=""/>
    <!-- Position where the object should be spawned. Format should be x y z (in meters) -->
    <arg name="object_position" default="0 0 0"/>
    <!-- Orientation of the object when spawned. Format should be r p y (in radians) -->
    <arg name="object_rpy_orientation" default="0 0 0"/>
    <!-- Name of the reference frame. It's according to this frame that the position and orientation are being set -->
    <arg name="reference_frame" default="world"/>
    <!-- Path to a sdf file describing the object to spawn (use that if the object is not in the description package) -->
    <arg name="file_path" default=""/>
    <!-- Launch the node -->
    <node name="manage_objects" pkg="grip_core" type="manage_object.py" output="screen" args="-d $(arg delete_object) -n '$(arg object_name)' -t '$(arg object_type)' -p '$(arg object_position)' -o '$(arg object_rpy_orientation)' -r '$(arg reference_frame)' -f '$(arg file_path)'"/>

</launch>
